community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/CypherAdapterStream.java

```diff
@@ -72,7 +80,7 @@ public boolean visit( Result.ResultRow row ) throws Exception
         } );
 
         QueryExecutionType qt = delegate.getQueryExecutionType();
-        visitor.addMetadata( "type", queryTypeCode( qt.queryType() ) );
+        visitor.addMetadata( "type", queryType );
```

Comment:
yes, I didn't touch that. I can remove it if that is what we think that is better? 

Just to clarify I was also talking about before streaming the result back. We know the `type` after parsing the query, before streaming the result so it should not be a problem of providing the `type` as a response to `RUN`